app/routes/downloadresults.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, ResponseToolkit, ResponseObject } from "@hapi/hapi";
+import Joi from "joi";
+import { getDownloadDetailsCsv } from "../backend/api";
+import { createModel } from "./models/searchResultsModel";
+const { Parser } = require('json2csv')
+
+module.exports = [
+  {
+    path: '/downloadresults',
+    method: 'GET',
+    options: {
+      auth: false,
+      validate: {
+        query: Joi.object({
+          searchString: Joi.string().trim().min(1).required(),
+          page: Joi.number().default(1),
+          pageId: Joi.string().default(''),
+          schemes: Joi.alternatives().try(Joi.string(), Joi.array()).default([]),
+          amounts: Joi.alternatives().try(Joi.string(), Joi.array()).default([]),
+          counties: Joi.alternatives().try(Joi.string(), Joi.array()).default([]),
+          sortBy: Joi.string().trim().optional().default('score'),
+        }),
+        failAction: async (request: Request, h: ResponseToolkit, error: any) => {
+          h.response(error.toString()).code(400).takeover()
+        }
+      },
+      handler: async (request: Request, h: ResponseToolkit) => {
+        const csvFields =  [
+          'payee_name',
+          'part_postcode',
+          'town',
+          'county_council',
+          'amount'
+        ]
+        request.query.action = 'download'
+        const model = await createModel(request.query)
```

Comment:
searchResultsModel's createModel usage is not ideal here because it does a lot of processing for filters and tags, so we are redundantly generating all of this data only to discard it later. Since we are only interested search part of the model, it would be more efficient to just copy some of the code you need in this handler. i.e, performSearch function and the code you need to call this function.

Reply:
Agreed, and updated the code accordingly.